src/Wizard.js

```diff
@@ -10,30 +10,52 @@ export class ValidationError extends Error {}
 type Props = {
   onComplete: (currentStep: string) => void,
   children: React$Node,
-  debug?: Boolean,
+  debug?: boolean,
+  updateUrl?: boolean,
 };
 
-const Wizard = ({ children, onComplete, debug }: Props) => {
+const Wizard = ({ children, onComplete, updateUrl, debug }: Props) => {
   const [index, setIndex] = useState(0);
   const [steps, setSteps] = useState([]);
   const [isLoading, setLoadingState] = useState(false);
 
+  const href = typeof window === 'undefined' ? '' : window.location.href;
+
   function registerStep(step) {
     const alreadyRegistered = steps.map(el => el.name).includes(step.name);
     if (!alreadyRegistered) {
-      setSteps(prevSteps => [...prevSteps, step]);
+      setSteps(previousSteps => [...previousSteps, step]);
     }
   }
 
   function updateStep(step) {
     const stepIndex = steps.findIndex(el => el.name === step.name);
-    setSteps(prevSteps => [
-      ...prevSteps.slice(0, stepIndex),
+    setSteps(previousSteps => [
+      ...previousSteps.slice(0, stepIndex),
       step,
-      ...prevSteps.slice(stepIndex + 1),
+      ...previousSteps.slice(stepIndex + 1),
     ]);
   }
 
+  function updateHistory(currentStepName, nextStepName) {
+    if (typeof window === 'undefined') {
+      return;
+    }
+    const currentUrl = window.location.href;
+    const basePath = currentUrl.includes('?')
+      ? currentUrl.split('?')[0]
+      : currentUrl;
+    const searchParams = new URLSearchParams(new URL(currentUrl).searchParams);
+    searchParams.set('step', nextStepName);
```

Comment:
Note to self: the search parameter key could probably be passed in as a prop from the consumer.